packages/delegation-process/test/approveDelegation.test.ts

```diff
@@ -0,0 +1,122 @@
+/* eslint-disable functional/no-let */
+import {
+  decodeProtobufPayload,
+  getMockDelegationProducer,
+  getMockTenant,
+} from "pagopa-interop-commons-test/index.js";
+import { describe, expect, it, vi } from "vitest";
+import {
+  DelegationApprovedV2,
+  DelegationId,
+  toDelegationV2,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { delegationState } from "pagopa-interop-models";
+import {
+  delegationNotFound,
+  operationRestrictedToDelegate,
+  incorrectState,
+} from "../src/model/domain/errors.js";
+import {
+  addOneDelegation,
+  delegationProducerService,
+  readLastDelegationEvent,
+} from "./utils.js";
+
+describe("approve delegation", () => {
+  it("should approve delegation in the happy path", async () => {
```

Comment:
using "happy path" sound too generic, I don't remember a description like this.
I suggest something like this:
```suggestion
  it("should approve delegation if all validation successed", async () => {
```